packages/api/src/lib/prepareParams.ts

```diff
@@ -28,7 +38,40 @@ function isEmpty(obj: any) {
   return [Object, Array].includes((obj || {}).constructor) && !Object.entries(obj || {}).length;
 }
 
-function processFile(paramName: string, file: string | ReadStream) {
+function isObject(thing: any) {
+  if (thing instanceof stream.Readable) {
+    return false;
+  }
+
+  return typeof thing === 'object' && thing !== null && !Array.isArray(thing);
+}
+
+function isPrimitive(obj: any) {
+  return typeof obj === null || typeof obj === 'number' || typeof obj === 'string';
+}
+
+function merge(src: any, target: any) {
+  if (Array.isArray(target)) {
+    // @todo we need to add support for merging array defaults with array body/metadata arguments
+    return target;
+  } else if (!isObject(target)) {
+    return target;
```

Comment:
Because we let you supply primitive values or a readable stream as a body argument we can't merge that with anything so we're dumping out and just returning whatever the user supplied.